lib/rust/parser/tests/parse.rs

```diff
@@ -36,18 +37,49 @@ macro_rules! block {
 // === Tests ===
 // =============
 
+#[test]
+fn nothing() {
+    test("", block![]);
+}
+
 #[test]
 fn application() {
     test("a b c", block![(App (App (Ident a) (Ident b)) (Ident c))]);
 }
 
 #[test]
-fn type_definition_bool() {
-    test("type Bool", block![(TypeDef (Ident type) (Ident Bool) #())]);
+fn parentheses_simple() {
+    let expected = block![
+        (MultiSegmentApp #(((Symbol "(") (App (Ident a) (Ident b))) ((Symbol ")") ())))
+    ];
+    test("(a b)", expected);
+}
+
+#[test]
+fn section_simple() {
+    let expected_lhs = block![(OprSectionBoundary (OprApp () (Ok "+") (Ident a)))];
+    test("+ a", expected_lhs);
+    let expected_rhs = block![(OprSectionBoundary (OprApp (Ident a) (Ok "+") ()))];
+    test("a +", expected_rhs);
 }
 
 #[test]
-fn type_definition_option() {
+#[ignore] // FIXME. This test catches a pre-existing bug.
```

Comment:
1. Fix left in the code
2. The fix does not explain anything.

Reply:
Fixed it (see below).